lib/bundler/process_lock.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module Bundler
+  class ProcessLock
+    def self.lock(bundle_path = Bundler.bundle_path)
+      lock_file_path = File.join(bundle_path, "bundler.lock")
+      File.open(lock_file_path, "w") do |f|
+        f.flock(File::LOCK_EX)
```

Comment:
Do you now when it is not supported, in the reading I did it was said that it would work even under Windows.